### PR TITLE
New package: Mathics

### PR DESCRIPTION
**Pull request by benley**
Mathics is a really cool alternative to Mathematica for computational algebra.

Updated a few needed dependencies.

### CODE REVIEW

**Comment by benley** [comment]
this appears to have caused a rebuild of Chromium.  FUN FUN FUN

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -7026,6 +7026,23 @@ let
...
+    name = "interruptingcow-${version}";
+    version = "0.6";
+
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/i/interruptingcow/${name}.tar.gz";
+      sha256 = "1cv4pm2h0f87n9w4r3l1f96skwmng95sawn7j00ns0rdp1zshr9d";
+    };
+
+    meta = {
+      description = "A watchdog that interrupts long running code.";
```
Comment (jagajaga) [inline pkgs/top-level/python-packages.nix]: Please remove dot at the end.

Comment (benley) [inline pkgs/top-level/python-packages.nix]: done.

**Comment by bjornfor** [comment]
@benley: Building mathics on my nixos system (with chroot) fails in the test phase:

```
$ nix-build -A mathics
[...]
======================================================================
ERROR: testPrompt (test_console.ConsoleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python2.7-mathics-0.8.drv-0/Mathics-v0.8-src/test/test_console.py", line 46, in testPrompt
    cons.expect('Quit by pressing CONTROL-D\r\n\r\n')
  File "/nix/store/hw167pv3hc4c2hyd0mkd8h4wli2lvvy1-python2.7-pexpect-3.3/lib/python2.7/site-packages/pexpect/__init__.py", line 1451, in expect
    timeout, searchwindowsize)
  File "/nix/store/hw167pv3hc4c2hyd0mkd8h4wli2lvvy1-python2.7-pexpect-3.3/lib/python2.7/site-packages/pexpect/__init__.py", line 1466, in expect_list
    timeout, searchwindowsize)
  File "/nix/store/hw167pv3hc4c2hyd0mkd8h4wli2lvvy1-python2.7-pexpect-3.3/lib/python2.7/site-packages/pexpect/__init__.py", line 1554, in expect_loop
    raise EOF(str(err) + '\n' + str(self))
EOF: End Of File (EOF). Exception style platform.
<pexpect.spawn object at 0x7fffef467310>
version: 3.3
command: /nix/store/zrny7m8nmn9q9klrr32alyfkkf113wc8-python-2.7.10/bin/python2
args: ['/nix/store/zrny7m8nmn9q9klrr32alyfkkf113wc8-python-2.7.10/bin/python2', 'mathics/main.py', '--color', 'NOCOLOR']
searcher: <pexpect.searcher_re object at 0x7fffef467ad0>
buffer (last 100 chars): ''
before (last 100 chars): 'mathics.core.definitions import Definitions\r\nImportError: No module named mathics.core.definitions\r\n'
after: <class 'pexpect.EOF'>
match: None
match_index: None
exitstatus: None
flag_eof: True
pid: 366
child_fd: 3
closed: False
timeout: 30
delimiter: <class 'pexpect.EOF'>
logfile: None
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0.05
delayafterclose: 0.1
delayafterterminate: 0.1

======================================================================
FAIL: testLaunch (test_console.ConsoleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python2.7-mathics-0.8.drv-0/Mathics-v0.8-src/test/test_console.py", line 22, in testLaunch
    cons.readline(), 'Mathics \\d\\.\\d.*\r\n')
AssertionError: Regexp didn't match: 'Mathics \\d\\.\\d.*\r\n' not found in '  File "mathics/main.py", line 27, in <module>\r\n'

----------------------------------------------------------------------
Ran 53 tests in 2.098s

FAILED (failures=1, errors=1, expected failures=5)
builder for ‘/nix/store/n9jqdbksy82isajqbknvh0mxv382kggf-python2.7-mathics-0.8.drv’ failed with exit code 1
error: build of ‘/nix/store/n9jqdbksy82isajqbknvh0mxv382kggf-python2.7-mathics-0.8.drv’ failed
```

**Comment by bjornfor** [comment]
```ImportError: No module named mathics.core.definitions```. Maybe it needs itself in $PYTHONPATH at test-time?

**Comment by benley** [comment]
Damn, I thought I saw the tests pass when I built it at home.  I'll try to get it fixed.

**Pull request merged**